client/me/account/toggle-sites-as-landing-page.tsx

```diff
@@ -0,0 +1,53 @@
+import { ToggleControl } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { successNotice } from 'calypso/state/notices/actions';
+import { savePreference } from 'calypso/state/preferences/actions';
+import { getPreference } from 'calypso/state/preferences/selectors';
+
+type ToggleSitesLandingPageProps = {
+	onToggle?: ( checked: boolean ) => void;
+	className?: string;
+};
+
+function ToggleSitesAsLandingPage( { onToggle, className }: ToggleSitesLandingPageProps ) {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const useSitesAsLandingPage = useSelector( ( state ) =>
+		getPreference( state, 'sites-landing-page' )
+	)?.useSitesAsLandingPage;
```

Comment:
It's usually better to select the data you need _inside_ the `useSelector` callback. That way it only re-renders when the specific preference you're interested in changes, not when any preference changes.

```suggestion
	const useSitesAsLandingPage = useSelector( ( state ) =>
		getPreference( state, 'sites-landing-page' )?.useSitesAsLandingPage
	);
```

More here under "Select the data you need inside the callback": p7H4VZ-2Q6-p2 (this P2 is talking about GB's `useSelect` specifically, but the same principle applies).

Reply:
TIL: thank you for that, I was actually following your [PR](https://github.com/Automattic/wp-calypso/pull/70356) but missed the fact that it was on the `getPreference`  method and not on the selector 😅

Reply:
Yeah this particular one is sorta easy to miss! Bracket-blindness or something like that 😅